File: water-bottles.ts
{
    const numWaterBottles = function(numBottles, numExchange) {
        let drink = numBottles;
    
        while (numBottles >= numExchange) {
            const fullWater = ~~(numBottles / numExchange);
    
            drink += fullWater;
            numBottles = fullWater + (numBottles % numExchange);
        }
        return drink;
    };

    console.log(numWaterBottles(9,3))
}

File: water-bottles.rs
pub fn num_water_bottles(mut num_bottles: i32, num_exchange: i32) -> i32 {
    let mut res = num_bottles;

    while num_bottles >= num_exchange {
        res += num_bottles / num_exchange;
        num_bottles = num_bottles / num_exchange + num_bottles % num_exchange;
    }

    res
}
